src/controllers/sdex/index.ts

```diff
@@ -29,12 +37,38 @@ router.get(
   }),
 );
 
-router.get('/user_pool_positions', async (req, res) => {
-  res.status(200).json({
-    data: {
-      user_pool_positions: [],
-    },
-  });
-});
+router.get(
+  '/user_pool_positions',
+  asyncRoute(async (req, res) => {
+    const { userId } = req.query;
+
+    if (!userId || typeof userId !== 'string') {
+      return res.status(400).json({ error: 'User ID is required' });
+    }
+
+    const variables = {
+      user: userId,
+    };
+
+    try {
+      const response = await req.network.sdex.queryFromSubgraph<LiquidityChangesResponse>(
+        LIQUIDITY_CHANGES_QUERY,
+        variables,
+      );
+      const userPoolPositions = response.liquidityChanges;
+      const aggregatedPositions = aggregatePositions(userPoolPositions);
+
+      return maybeCacheResponse(
+        res,
+        `sdex/user_pool_positions/${req.network.chainId}/${userId}`,
+        async () => aggregatedPositions,
+        DEFAULT_CACHE_TTL,
+      ).then((data) => res.json(toResponse(data)));
+    } catch (error) {
+      console.error('Error querying user pool positions:', error);
+      return res.status(500).json({ error: 'Failed to fetch user pool positions' });
```

Comment:
Use helper classes for custom errors (can find list of them in custom-error.ts file):

```suggestion
      throw new HttpError(500, error.message);
```

You also dont actually need try/catch in the controllers - `asyncRoute` middleware will handle errors and should print callstack to console automatically.

Btw, use `logger.` helper for logs instead of `console.`